### PR TITLE
Refactor accordion component

### PR DESCRIPTION
**Pull request by AlexOverbeck**
## Deprecation Notice:
The `title` @input has been deprecated in favor of the heading @input. Furthermore, the heading @input will no longer accept html and will only output a string value.

#### The reasons for this are:
1. This avoids the collision with the html native `title` attribute.
2. No longer accepting html ensures a more consistent implementation.

#### Refactors accordion component
This commit turns back on view encapsulation and refactors the
accordion component. The most notable changes are the removal of the
accordion template and reworking of all of the styles to correspond
with the components they apply to.

#### Upgrades the jasmine module
There wasn't any reason for us not to be on the newest version of jasmine.
This update gives us access to a lot of functionality that we didn't have
previously.

#### Refactors and tests the go accordion and panel components
I am finally getting the hang of this testing thing. This PR includes pretty
extensive testing of the accordion component and how it interacts with
the accordion panel component.

#### Adds dark theme back in to accordion
This allows panels to be displayed in dark mode for use on dark
backgrounds. It also updates the accordion to honor all of the
attributes explicitly set on the panel. This allows individual
panels to override the attributes defined by the accordion.

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-accordion/go-accordion-panel.component.html [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
-      <span class="go-accordion-panel__title-text" [innerHtml]="title"></span>
-    </span>
-    <span class="go-accordion-panel__control">
-      <span class="material-icons go-accordion-panel__control-icon">expand_more</span>
+      <go-icon
+        *ngIf="icon"
+        [icon]="icon"
+        class="go-accordion-panel__title-icon"
+      ></go-icon>
+      <h2 class="go-accordion-panel__title-text">{{ heading }}</h2>
```
[the remark this answers is not in the document]
Comment (AlexOverbeck) [reply projects/go-lib/src/lib/components/go-accordion/go-accordion-panel.component.html]: Yeah, I wasn't sure. I figured it was somewhere around secondary information. It seems like maybe these should also be one of our main header styles. What do you think?

[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-accordion/go-accordion-panel.component.html]: @AlexOverbeck yeah, maybe. What about just using the `<header>` element? It's specific that it's a header to the section below it, but doesn't assume any of hierarchy?

[the remark this answers is not in the document]
Comment (AlexOverbeck) [reply projects/go-lib/src/lib/components/go-accordion/go-accordion-panel.component.html]: @grahamhency I did some research here. There is a lot of flexibility here as [defined by the W3C](https://www.w3.org/TR/2014/REC-html5-20141028/sections.html#the-h1,-h2,-h3,-h4,-h5,-and-h6-elements).

From what I am reading here, this `<header>` and the heading inside of it represent a specific section document. To me, this means that we should change it to an `<h1>` as it is the most important heading in the `<header>` and maybe we should add a `<section>` below it.

[the remark this answers is not in the document]
Comment (AlexOverbeck) [reply projects/go-lib/src/lib/components/go-accordion/go-accordion-panel.component.html]: I read more about this, I was _almost_ right. I think this is the most appropriate solution:
```html
<section>
  <header>
    <h1>The header for the content</h1>
  </header>
  <div>
    <p>The panels content.</p>
  </div>
</section>
```

**Pull request merged**